src/main/kotlin/org/team5419/frc2019offseason/controllers/TeleopController.kt

```diff
@@ -67,68 +83,74 @@ public class TeleopController(
     override fun update() {
         // Driver
         isSlow = false
-        speed = Input.BASE_SPEED
-        if (mDriver.getAButton()) { isReverse = !isReverse }
-        if (isReverse) { speed *= -1 }
-        if (mDriver.getBumper(Hand.kLeft) || mDriver.getBumper(Hand.kLeft)) {
-            speed *= Input.SLOW
-        }
-
-        mSubsystems.drivetrain.setPercent(
-            driveHelper.calculateOutput(
-                mDriver.getY(Hand.kLeft) * speed,
-                mDriver.getX(Hand.kLeft),
-                isQuickTurn,
-                isHighGear
-            )
+        speed = 0.0
+        if (mDriver.getAButton()) { mDrivetrain.isReverse = !mDrivetrain.isReverse }
+        if (mDriver.getBumper(Hand.kLeft)) { mDrivetrain.isSlow = true } else mDrivetrain.isSlow = false
+
+        val ds: DriveSignal = driveHelper.calculateOutput(
+            mDriver.getY(Hand.kLeft),
+            mDriver.getX(Hand.kLeft),
+            isQuickTurn,
+            isHighGear
         )
+        // println("$ds.left $ds.right")
+        mSubsystems.drivetrain.setPercent(ds)
 
         // Climb control
         if (mDriver.getBumperPressed(Hand.kRight) || mCoDriver.getBumperPressed(Hand.kLeft)) {
+            if (isUnlocked == false) {
+                mSubsystems.climber.unlock()
+                isUnlocked = true
+            }
             mSubsystems.climber.climb()
         }
 
         // Codriver
         // Valve control
         if (mCoDriver.getBumperPressed(Hand.kLeft) || mCoDriver.getBumperPressed(Hand.kRight))
             mSubsystems.vacuum.release()
+
         // Vacuum control
         if (mCoDriver.getTriggerAxis(Hand.kLeft) > Input.DEADBAND) {
-            mSubsystems.vacuum.pickBall(mCoDriver.getTriggerAxis(Hand.kLeft))
+            mVacuum.pickBall(mCoDriver.getTriggerAxis(Hand.kLeft))
         } else if (mCoDriver.getTriggerAxis(Hand.kRight) > Input.DEADBAND) {
-            mSubsystems.vacuum.pickHatch(mCoDriver.getTriggerAxis(Hand.kRight))
+            mVacuum.pickHatch(mCoDriver.getTriggerAxis(Hand.kRight))
         } else { mSubsystems.vacuum.setPercent(0.0) }
+
         // button control
         if (mCoDriver.getAButtonPressed()) {
-            mSubsystems.lift.setPosistion(LiftHeight.HATCH_LOW)
+            mLift.setPosistion(LiftHeight.HATCH_LOW)
         } else if (mCoDriver.getBButtonPressed()) {
             mSubsystems.lift.setPosistion(LiftHeight.HATCH_MID)
-        } else if (mCoDriver.getYButtonPressed())
+        } else if (mCoDriver.getYButtonPressed()) {
             mSubsystems.lift.setPosistion(LiftHeight.HATCH_HIGH)
-        else if (mCoDriver.getPOV() == 0) {
+        } else if ((mCoDriver.getPOV() >= 340 && mCoDriver.getPOV() <= 360) ||
```

Comment:
change this to `} else if ((mCoDriver.getPOV() >= 340 || mCoDriver.getPOV() <= 20)